updater/lib/dependabot/updater/operations/update_all_versions.rb

```diff
@@ -0,0 +1,357 @@
+# frozen_string_literal: true
+
+module Dependabot
+  class Updater
+    module Operations
+      class UpdateAllVersions
+        def self.applies_to?(job:)
+          return false if job.security_updates_only?
+          return false if job.updating_a_pull_request?
+          return false if job.dependencies&.any?
```

Comment:
Yeah, for a fresh Version Update, we never pre-fill the `dependencies` array, it is only prepopulated for Security-Updates or rebases.

If we were to do "fresh", targeted Version Updates with a pre-filled set of `dependencies` with the existing code I think they might "work by accident" - we've had some experiments around 'click to update this one Dependency' in the past that benefited from this.

I think the spirit of this Operation refactor is that we should not have a strategy for a scenario we currently do not explicitly support, so essentially if we wanted to do "fresh", targeted Version Updates in future, we would define a new class with:

```
def self.applies_to?(job:)
  return false if job.security_updates_only?
  return false if job.updating_a_pull_request?
  return false if job.dependencies.empty?
end
```

It would then implement a version of this logic that _explicitly_ resolved the job.dependencies with the parsed Dependencies.